include/git2/proxy.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_proxy_h__
+#define INCLUDE_git_proxy_h__
+
+#include "common.h"
+#include "transport.h"
+
+GIT_BEGIN_DECL
+
+/**
+ * The type of proxy to use.
+ */
+typedef enum {
+	/**
+	 * Do not attempt to connect through a proxy
+	 */
+	GIT_PROXY_NONE,
+	/**
+	 * Try to auto-detect the proxy from the git configuration.
+	 */
+	GIT_PROXY_AUTO,
+	/**
+	 * Connect through a HTTP proxy
+	 */
+	GIT_PROXY_HTTP,
+	/**
+	 * Connect through a SOCKS v4 proxy
+	 */
+	GIT_PROXY_SOCKS4,
+	/**
+	 * Connect through a SOCKS v5 proxy
+	 */
+	GIT_PROXY_SOCKS5,
+} git_proxy_t;
+
+/**
+ * Options for connecting through a proxy
+ *
+ * Note that not all types may be supported, depending on the platform
+ * and compilation options.
+ */
+typedef struct {
+	unsigned int version;
+
+	/**
+	 * The type of proxy to use, by URL, auto-detect.
+	 */
+	git_proxy_t type;
+
+	/**
+	 * The URL of the proxy.
+	 */
+	const char *url;
+
+	/**
+	 * The port of the proxy
+	 */
+	const char *port;
```

Comment:
We should probably delete it and just have the url for the whole thing. When we use our own connection functions, we split it out, but IIRC both WinHTTP and curl will happily take it as a url string.